### ISSUE TITLE
`Ctrl` with `Tab`/`Shift`+`Tab` should work in `ToolStrip`

### ISSUE THREAD

**Issue by ArtemTatarinov** [dotnet/winforms#5752]
- `Ctrl`+`Tab` should move button focus forward inside `ToolStrip` when `TabStop` is `false`
- `Ctrl`+`Shift`+`Tab` should move button focus backward inside `ToolStrip`

This is a problem. Ctrl+Tab should move "forward", Ctrl+Shift+Tab should move backwards . Try office toolstrips. Consider a user who navigates office with one set of keys, even unintentionally, and a winforms app with another. Sounds like an annoying experience. I always want to hit f12 to go to source in GH web UI for example. Even worse if office ever works with Core plugins....

_Originally posted by @Tanya-Solyanik in https://github.com/dotnet/winforms/pull/5726#discussion_r706427053_

**Comment by weltkante** [comment]
The implementation should be done in a way that its possible to opt-out of this behavior if the application wants to give this shortcut a different meaning yet still have toolbars. Some multi-document applications prefer to switch documents with those shortcuts.

**Comment by ArtemTatarinov** [comment]
It's possible to use `Ctrl` for moving `ToolStrip` button focus forward and backward with rather simple modification.

**`ProcessDialogKey` method at `ToolStrip` class**

Before the fix:
```
bool hasModifiers = ((keyData & (Keys.Alt | Keys.Control)) != Keys.None);
...
case Keys.Tab:
    if (!hasModifiers)
    {
        retVal = ProcessTabKey((keyData & Keys.Shift) == Keys.None);
    }
```

After the fix:
```
case Keys.Tab:
    if ((keyData & Keys.Alt) == Keys.None)
    {
       retVal = ProcessTabKey((keyData & Keys.Shift) == Keys.None);
    }
```
- `hasModifiers` is used at other switch cases, so we couldn't get rid of it completely
- All `Alt`+`Tab`/`Alt`+`Ctrl`+`Tab`/`Alt`+`Shift`+`Tab `/`Alt`+`Ctrl`+`Shift`+`Tab` combinations are intercepted by Windows, but we can be cautious and make an additional check for `Alt` not being pressed here.

- `ProcessDialogKey` method is virtual, so the custom application could override it and use its own `Tab` press handling. In addition to the fact that focus moving between the `ToolStrip` buttons occurs only if its `TabStop` is `false` and we already have focus at some of the `ToolBar` buttons (like `ToolStripTextBox`, for example), maybe it would sufficient for possible opting-out of moving focus with `Ctrl`+`Tab`?

**Comment by weltkante** [comment]
> ProcessDialogKey method is virtual, so the custom application could override it and use its own Tab press handling.

Thats not necessarily sufficient if the code that needs to be called isn't accessible, but for this case I think you could just calle `base.ProcessDialogKey` if anything other than `Ctrl`+`Tab` is pressed, so probably ok.

**Comment by weltkante** [comment]
> ProcessDialogKey method is virtual, so the custom application could override it and use its own Tab press handling.

Thats not sufficient, you also need everything the original implementation called being public, which it apparently isn't. For example the implementation is calling `_parent.ProcessDialogKey(keyData)` which I think is not possible without doing reflection. You lose the ability to process `Ctrl`+`Tab` in a central place.

**Comment by ArtemTatarinov** [comment]
@weltkante, thank you for your answer!

@Tanya-Solyanik, could you please tell what's your opinion on this topic?

**Comment by NikitaSemenovAkvelon** [comment]
It works correctly if I don't misunderstand a problem. Ctrl+Tab shortcut moves focus to next ToolStrip and with Shift moves to previous one.  
![jRBHLJ132Q](https://user-images.githubusercontent.com/109065597/179750850-4938a389-0604-4a72-b0d2-907a41f2d903.gif)

@Olina-Zhang could you test it, please?
@dreddy-work, @merriemcgaw what do you think if we close this issue as not reproducing?

**Comment by dreddy-work** [comment]
@NikitaSemenovAkvelon , did you hit same code path that was in the comment here ?

**Comment by merriemcgaw** [comment]
@Tanya-Solyanik what are your thoughts on this one? The behavior looks reasonable to me but I don't have a full understanding of what you were trying to get done in the original comment.

**Comment by Tanya-Solyanik** [comment]
@merriemcgaw - in office ribbon Ctrl+Tab moves focus to the next item in the current toolstrip, not to the next toolsstrip.

**Comment by Olina-Zhang** [comment]
> @Olina-Zhang could you test it, please?

@NikitaSemenovAkvelon I have the same testing result as you. And Tanya has given the comment: when `TabStop` is false, `Ctrl+Tab `should move button focus forward **inside** ToolStrip, instead of between ToolStrips.

**Comment by NikitaSemenovAkvelon** [comment]
I've completed a small research about behavior of tabbing in few apps. There are some differences:
1) **Office**:
![ezgif-2-c7d72a6d53](https://user-images.githubusercontent.com/109065597/180189171-9addfb22-c771-4e3e-97a0-b1f29151c04b.gif)
![image](https://user-images.githubusercontent.com/109065597/180189277-a112962a-b1e5-4afa-94e2-8ad4835e6fe3.png)
Behavior of `tab` doesn't depend on `ctrl` modicator. But an important detail is that panel is not a ToolStrip but just a Group. You can see it in Inspect screen.
2) **Visual Studio**:
![image](https://user-images.githubusercontent.com/109065597/180191091-63fd7c7b-0f85-485a-824b-8d3b79489b2c.png)
![image](https://user-images.githubusercontent.com/109065597/180191358-bdc7ae45-f367-432e-8999-c3e061c8fc54.png)
Behavior of `tab` depends on `ctrl` modicator. And panel is a ToolBar.
3) **WinForms application**:
![IThOpSCylV](https://user-images.githubusercontent.com/109065597/180192385-acd8527a-5a0f-4675-b6d6-60a9cdb017ae.gif)
![image](https://user-images.githubusercontent.com/109065597/180192010-3b1fdc45-f567-4d43-acc6-b7404dfd0390.png)
First pair of panels has `TabStop = false`, second pair has `TabStop = true`. Behavior of the first pair is the same as Visual Studio, moreover type of panels is the same too. Second pair works differently: pressing `Tab` switches panels, but pressing `Ctrl+Tab` does nothing.
I've done all of them because I not completely understand what wrong with current behavior. For me it looks correctly. What do you think about this behavior in comparing with Office and VisualStudio, taking into account difference of controls?

**Comment by merriemcgaw** [comment]
It looks OK to me too. @Tanya-Solyanik, given that these are ToolStrips rather than Ribbons, my thought is that we should be closer to the behavior in VS. What are your thoughts?

**Comment by Tanya-Solyanik** [comment]
@NikitaSemenovAkvelon, @merriemcgaw , @Olina-Zhang  -  
Well, Office is the standard in accessibility. VS usage is miniscule compared to office's 😄 . So I would rather copy Office's experience. But this issue overall is a minor one because the user is not stuck within the control either way, there is always a way to exit it.  And I can see than Ctrl+Tab is a convenience when user wants to skip the toolstrip altogether. So I will not insist here.

**Comment by merriemcgaw** [comment]
The behavior of Tabbing through ToolStripContainers has always been a mystery to me 😆. I think we're in an OK place here, and that we've got better places to invest our time in 😉 . Until we find out that the current behavior is a MAS violation, I think we can close this issue.
